### PR TITLE
[Integration Tests] CM coverage for InvalidStakeDepth and MaxReorgViolation

### PR DESCRIPTION
**Pull request by fassadlr**
This contains a small change to `StakeValidator` so that we can set network consensus options in the `InvalidStakeDepth` integration test scenario.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs [hunk 1]
```diff
@@ -21,20 +23,18 @@ public ConsensusManagerTests()
...
             public override int GetStakeMinConfirmations(int height, Network network)
             {
-                return height < 10 ? 5 : 10;
+                return height < 15 ? 5 : 20;
             }
         }
 
         [Fact]
-        public void ConsensusManager_Fork_Occurs_MinerNode_Reorgs_AndResyncs_ToBestHeight()
+        public void ConsensusManager_Fork_Occurs_Node_Reorgs_AndResyncs_ToBestHeight()
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Can you just specify in the comment the scenario this covers?

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs [hunk 2]
```diff
@@ -71,16 +71,134 @@ public void ConsensusManager_Fork_Occurs_MinerNode_Reorgs_AndResyncs_ToBestHeigh
...
+                TestHelper.Disconnect(syncer, minerA);
+                TestHelper.Disconnect(syncer, minerB);
+
+                // Ensure syncer does not have any connections.
+                TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(syncer));
+
+                // MinerA continues to mine to height 45.
+                TestHelper.MineBlocks(minerA, walletName, walletPassword, walletAccount, 25);
+
+                // MinerA continues to mine to height 65.
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Change to `MinerB`

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs [hunk 3]
```diff
@@ -71,16 +71,134 @@ public void ConsensusManager_Fork_Occurs_MinerNode_Reorgs_AndResyncs_ToBestHeigh
...
+
+                Assert.True(syncer.FullNode.ConsensusManager().Tip.Height == 15);
+                Assert.True(minerB.FullNode.ConsensusManager().Tip.Height == 15);
+            }
+        }
+
+        [Fact]
+        public void ConsensusManager_Fork_Occurs_Node_Gets_Disconnected_Due_To_MaxReorgViolation()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Wehre do we set the max reorg violation to 15?

Comment (zeptin) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: It is `500` by default on StratisTest (passed into `Consensus` constructor when the `Network` is instantiated), but is being overridden at the top of this source file to `20`.

Note that there is also a `MaxReorgLength` in `ChainState` that should probably be removed, judging by various TODOs.

Comment (dangershony) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Yes sorry I meant I think in this test he is overriding it to something smaller, so just asking where this happens.

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Fixed 👍

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: Hey @dangershony, yes I am setting it via reflection in the constructor of the test class 👍

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs]: @dangershony This very basic re-org test doesn't specifically belong to any of the scenarios in Ivan/Apro's test document. Is this OK? I will start work on those specific tests after this is merged 👍

**Pull request merged**